kbn_pm/src/commands/bootstrap/bootstrap_command.mjs

```diff
@@ -98,30 +83,25 @@ export const command = {
       }),
     ]);
 
-    // Bootstrap process for Bazel packages
-    // Bazel is now managing dependencies so yarn install
-    // will happen as part of this
-    //
-    // NOTE: Bazel projects will be introduced incrementally
-    // And should begin from the ones with none dependencies forward.
-    // That way non bazel projects could depend on bazel projects but not the other way around
-    // That is only intended during the migration process while non Bazel projects are not removed at all.
-    if (forceInstall) {
-      await time('force install dependencies', async () => {
-        await removeYarnIntegrityFileIfExists();
-        await Bazel.expungeCache(log, { quiet });
-        await Bazel.installYarnDeps(log, { offline, quiet });
+    if (shouldInstall) {
```

Comment:
reworked / manualized the logic around here. 
There are a few final scenarios now:
 - `node_modules` missing: full reinstall irrespective of flags.
 - `node_modules` present, with `--force-install`: the yarn integrity file will be removed, that triggers a reinstall, but it's faster than a full reinstall with some modules present
 - `node_modules` present, with an out-of-date yarn integrity file: yarn will do a quick install, only fixing whatever is broken, probably
 - `node_modules` present, with an up-to-date yarn integrity file: only an integrity check is done, and module installs are skipped

Reply:
@delanni likely useful to add the above explanation into comments in the code